Point Information not displayed in the details Pane for "Polygon" layers

Tested Build:http://fgpv.cloudapp.net/demo/v1.5.0-5/prod/samples/index-fgp-en.html?keys=JOSM,EcoAction,CESI_Other,NPRI_CO,Railways,on_crop,Airports,Barley

Uploaded File : happy.json; canada.json

[happy.zip](https://github.com/fgpv-vpgf/fgpv-vpgf/files/666657/happy.zip)

There seems to be an issue while querying the Polygon Layer. Details Pane is empty even when the layer has data in it. Please investigate

![image](https://cloud.githubusercontent.com/assets/18539315/21393820/11d5f4ba-c764-11e6-8af7-74e1cc85d4a9.png)


![image](https://cloud.githubusercontent.com/assets/18539315/21393698/8d9997b0-c763-11e6-84b4-11edac7c8b4b.png)

Polygon features from added files (Shapefile, geoJson) generate this error 
`core.js:140115 Error: FeatureLayer::_query - query contains one or more unsupported parameters` 

As found [here](https://developers.arcgis.com/javascript/3/jsapi/featurelayer-amd.html#featurelayer2):

> Creates a new instance of a feature layer using a FeatureCollection object. The feature layer, when initialized with a feature collection object has the following behavior:
...
Does not support queries that need to be performed on the server, e.g. queries with a where clause or non-extent based spatial queries.
...

So, I think the non-extent is the problem with actual polygon query because we're using a point coordinates and not extent.

--- a/src/app/geo/identify.service.js
+++ b/src/app/geo/identify.service.js
@@ -353,7 +353,9 @@
                 qry.outFields = ['*']; // this will result in just objectid fields, as that is all we have in feature layers
 
                 // more accurate results without making the buffer if we're dealing with extents
-                if (layerRecord._layer.geometryType === 'esriGeometryPolygon') {
+                // polygons from added file need buffer
+                if (layerRecord._layer.geometryType === 'esriGeometryPolygon' &&
+                    legendEntry.id.search('esriFeature#') === -1) {
                     qry.geometry = opts.geometry;
                 } else {
                     qry.geometry = makeClickBuffer(opts.clickEvent.mapPoint, opts.map, layerRecord.clickTolerance);

